// vercel/dist/chunks/image-FGP5UVA5.js
import { createRequire as __createRequire } from 'node:module';
import { fileURLToPath as __fileURLToPath } from 'node:url';
import { dirname as __dirname_ } from 'node:path';
const require = __createRequire(import.meta.url);
const __filename = __fileURLToPath(import.meta.url);
const __dirname = __dirname_(__filename);
import {
  getInvalidSubcommand
} from "./chunk-VGIMO3ZK.js";
import {
  getSubcommand
} from "./chunk-YPQSDAEW.js";
import {
  AGENT_REASON
} from "./chunk-IC4YEIGW.js";
import {
  buildCommandWithGlobalFlags,
  outputAgentError
} from "./chunk-CB3I3QIT.js";
import "./chunk-P6AK7SVK.js";
import "./chunk-P4QNYOFB.js";
import {
  output_manager_default
} from "./chunk-OX7KI3LF.js";
import "./chunk-S7KYDPEM.js";
import "./chunk-TZ2YI2VH.js";

// src/commands/vcr/image/index.ts
var IMAGE_CONFIG = {
  ls: ["ls", "list"],
  inspect: ["inspect", "get"],
  rm: ["rm", "remove", "delete"]
};
async function image(client, argv, telemetry) {
  const { subcommand, args } = getSubcommand(argv, IMAGE_CONFIG);
  if (subcommand == null) {
    const message = argv.length === 0 ? getInvalidSubcommand(IMAGE_CONFIG) : `Unknown "vcr image" subcommand "${argv[0]}".`;
    outputAgentError(
      client,
      {
        status: "error",
        reason: AGENT_REASON.INVALID_ARGUMENTS,
        message,
        next: [
          {
            command: buildCommandWithGlobalFlags(
              client.argv,
              "vcr image --help"
            ),
            when: "Show valid image subcommands"
          }
        ]
      },
      1
    );
    output_manager_default.error(`${message} Run \`vercel vcr image --help\`.`);
    return 1;
  }
  switch (subcommand) {
    case "ls":
      return (await import("./ls-EKNQN3V4.js")).default(client, args, telemetry);
    case "inspect":
      return (await import("./inspect-N5DHCKLZ.js")).default(client, args, telemetry);
    case "rm":
      return (await import("./rm-SD3EJRXY.js")).default(client, args, telemetry);
    default:
      output_manager_default.error(`Unhandled image subcommand: ${subcommand}`);
      return 1;
  }
}
export {
  image as default
};
